stacks-node/src/burnchains/rpc/bitcoin_rpc_client/test_utils.rs

```diff
@@ -0,0 +1,183 @@
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Test-only utilities for [`BitcoinRpcClient`]
+
+use serde_json::Value;
+
+use crate::burnchains::rpc::bitcoin_rpc_client::{BitcoinRpcClient, BitcoinRpcClientResult};
+
+/// Represents the response returned by the `getblockchaininfo` RPC call.
+///
+/// # Notes
+/// This struct supports a subset of available fields to match current usage.
+/// Additional fields can be added in the future as needed.
+#[derive(Debug, Clone, Deserialize)]
+pub struct GetBlockChainInfoResponse {
+    /// the network name
+    pub chain: String,
+    /// the height of the most-work fully-validated chain. The genesis block has height 0
+    pub blocks: u64,
+    /// the current number of headers that have been validated
+    pub headers: u64,
+    /// the hash of the currently best block
+    #[serde(rename = "bestblockhash")]
+    pub best_block_hash: String,
+}
+
+/// Represents the response returned by the `generateblock` RPC call.
+#[derive(Debug, Clone, Deserialize)]
+struct GenerateBlockResponse {
+    /// The hash of the generated block
+    hash: String,
+}
+
+impl BitcoinRpcClient {
+    /// Retrieve general information about the current state of the blockchain.
+    ///
+    /// # Arguments
+    /// None.
+    ///
+    /// # Returns
+    /// A [`GetBlockChainInfoResponse`] struct containing blockchain metadata.
+    pub fn get_blockchain_info(&self) -> BitcoinRpcClientResult<GetBlockChainInfoResponse> {
+        Ok(self
+            .global_ep
+            .send(&self.client_id, "getblockchaininfo", vec![])?)
+    }
+
+    /// Retrieves the raw hex-encoded transaction by its ID.
+    ///
+    /// # Arguments
+    /// * `txid` - Transaction ID (hash) to fetch.
+    ///
+    /// # Returns
+    /// A raw transaction as a hex-encoded string.
+    ///
+    /// # Availability
+    /// - **Since**: Bitcoin Core **v0.7.0**.
+    pub fn get_raw_transaction(&self, txid: &str) -> BitcoinRpcClientResult<String> {
+        Ok(self
+            .global_ep
+            .send(&self.client_id, "getrawtransaction", vec![txid.into()])?)
+    }
+
+    /// Mines a new block including the given transactions to a specified address.
+    ///
+    /// # Arguments
+    /// * `address` - Address to which the block subsidy will be paid.
+    /// * `txs` - List of transactions to include in the block. Each entry can be:
+    ///   - A raw hex-encoded transaction
+    ///   - A transaction ID (must be present in the mempool)
+    ///   If the list is empty, an empty block (with only the coinbase transaction) will be generated.
+    ///
+    /// # Returns
+    /// The block hash of the newly generated block.
+    ///
+    /// # Availability
+    /// - **Since**: Bitcoin Core **v22.0**.
+    /// - Requires `regtest` or similar testing networks.
+    pub fn generate_block(&self, address: &str, txs: &[&str]) -> BitcoinRpcClientResult<String> {
```

Comment:
Please use `BurnchainHeaderHash` instead of `String` for representing a Bitcoin block hash.